### DISCUSSION TITLE
Is Beaker still being worked on?

### DISCUSSION THREAD

**Discussion by ghost** [beakerbrowser/beaker#1944] [Q&A] [answered]
I see the last commit was just over 10 months ago, will Beaker ever be updated again or do the maintainers consider it to be "done" now? I really enjoy using Beaker so it'd suck if it was dead. gosh bless

**Comment by pfrazee** [comment, marked as answer]
Yeah I'm real sorry to say, I'm going to be closing down this project in the very near future. I'll do a post mortem and share more, but in the intro to my follow-on project I do give a few words on why I felt it was time to change the approach: https://atek.cloud/blog/hello-world

**Reply by ghost** [reply]
truly a bruh moment

**Reply by ghost** [reply]
(atek look nice, though)

**Reply by pfrazee** [reply]
Yeah I really hate disappointing folks like that. FWIW, Atek is very much trying to be everything that Beaker was (and more). Also, Brave and Opera are experimenting with IPFS in the same way Beaker experimented with Hypercore, so give them a look.

**Reply by RangerMauve** [reply]
If y'all are still interested in web browsers, you might wanna check out Agregore. 😁  https://agregore.mauve.moe/

**Reply by xldrkp** [reply]
@pfrazee Great that you have new ideas! I'm planning to give a hands-on introduction to the dweb with Beaker. Will you keep the website and the download online for a while?

**Reply by pfrazee** [reply]
I will! I second Ranger's recommendation of Agregore. Also Brave and Opera are experimenting with IPFS. But! Beaker will remain available for a while to at least demo some of the crazy ideas 😂

**Reply by xldrkp** [reply]
Wonderful! I love to hear that. I checked out Agregore yesterday. It's promising as it embraces the diversity of federated protocols. What I wish for is an editing function like in Beaker which is crucial for use in web app education.

**Reply by RangerMauve** [reply]
I actually did a workshop using chrome dev tools to edit content and save it back into the hyperdrive. Feel free to email me if you wanna reuse my workshop materials or want help developing one. contact@mauve.moe

https://blog.mauve.moe/slides/agregore-blog-workshop/#1

**Reply by SandroLinux** [reply]
@pfrazee but where will we host our Hyper websites?

**Reply by Kyllingene** [reply]
@SandroLinux Isn't Hypercore P2P, so there's no central hosting server to shut down?

**Reply by winstonford** [reply]
@pfrazee Oh wow, just catching wind of this.. after I pulled a git repo straight offa gitlab into a local hyperdrive!  So I read the Atek blog post.. and I jus want to state for the record, that no one, and I mean no one has ever even heard of raspberry pi, let alone has or will ever have one.  Now, I am not talking about myself or others who have spent countless hours with pi's and beaglebone blacks and diskutil in the days before resin.io ported docker to arm and released Etcher.io and then changed its named to balena.io... nooo I am talking about the general public.  Everyone has heard of Safari or Chrome or IE, and has one.  Everyone has a phone with one of these browsers on it.  They might even have a home cloud in the form of Apple TV or Alexa.. but as much as I will most likely try and get the atek home cloud running on a pi zero.. no matter if you achieve excellence beyond your wildest dreams with atek.. you will be limited to the minuscule user base of pi.  For this p2p web to really happen, its needs to run on phones.  Everyone has a phone, and as much time as they are active.. they are also inactive and could be hosting the p2p web.  Paul, thanks for all the beaker work and I wish you and atek the best.. But please consider taking beaker to iOS / android.  I dont know much about electron and chromium, but there must be a way!  Ps. In the roadmap https://atek.cloud/docs/manual/dev/roadmap I now see iOS and android :)

**Reply by Kyllingene** [reply]
This probably isn't the place for that, but either way I feel you stated that rather more strongly than warranted. Raspberry Pi is gradually becoming more mainstream (i.e. pi 400), and while nowhere near as widespread as phones or desktops, it really isn't fitting to say "no one has ever even heard of raspberry pi." Also, there are many reasons why basing a p2p web on phones is not feasible; they are (by definition) mobile devices, and thus could go anywhere; you can never guarantee where it is or who its neighbors are, and making a p2p service off of that won't work.

Also, you need computers in the loop anyway; nobody's hosting a website on their phone. And what about in more rural areas, where big service providers already have problems providing consistent internet? What about those people whose connections would be through one phone, that may or may not be on or nearby at any given time? There are simply too many issues with basing an entire p2p web on mobile to expect that to be their primary concern.

**Reply by RangerMauve** [reply]
Not to distract too much from Beaker stuff, but I'm actually putting together a grant application to make a mobile version of the [Agregore Browser](https://agregore.mauve.moe/) some time this summer. 😁 If anyone knows C++ devs that'd be interested in a few months of work, hit me up.

We're going to aim for working in low-connectivity areas such as community mesh networks in the global south, and mobile seems to be the main way folks are doing computing out there. Though we're also going to work with the [Nimble](https://wakoma.co/nimble/) project which lets communities run services locally on their mesh node which is similar to the Atek concept (but more manual at the moment).

**Reply by SandroLinux** [reply]
Nice :). Sound awesome, will help in whatever way I can (but I don't know C++ that well)

**Reply by SandroLinux** [reply]
> @SandroLinux Isn't Hypercore P2P, so there's no central hosting server to shut down?

I am saying what other software is there to host beaker sites?

**Reply by winstonford** [reply]
Interesting thread here!  Last night @Kyllingene got me thinking with the statement "nobody's hosting a website on their phone." Well, low and behold.. someone IS hosting a website on their phone!  Thx to the wonders of https://ish.app I am hosting my static website on my iPhone 5s... as of right now.  It's only accessible on my local network.  As far as I can tell, cloudflared, the tunneling app, has some problems working on Alpine linux.. so I am considering a reverse ssh tunnel to let the site be reachable from the internet.  The process to get the site hosted on the phone is wicked simple:  Install the python web server, wget my static.tar.gz from the internet, unzip, start the server and BLAM accessible at http://10.0.0.165:8000 is my site.  After installing ssh on the iPhone, in ish, I can now ssh into it from my MacBook.. which speeds up dev quite a bit.  On the iphone5s, alpine is a bit laggy over ssh, but the websever is surprisingly snappy. On my iphone12, ish and alpine is pretty dang fast.  If in anyone is interested, lemme know.  I just followed the docs here: https://github.com/ish-app/ish/wiki

**Reply by RangerMauve** [reply]
> 

I use running on a Digital Ocean Droplet (5$/month) and its been working pretty well for a few years now. 😁 https://github.com/datproject/dat-store

**Reply by ceewanna** [reply]
> Not to distract too much from Beaker stuff, but I'm actually putting together a grant application to make a mobile version of the [Agregore Browser](https://agregore.mauve.moe/) some time this summer. grin If anyone knows C++ devs that'd be interested in a few months of work, hit me up.
> 
> We're going to aim for working in low-connectivity areas such as community mesh networks in the global south, and mobile seems to be the main way folks are doing computing out there. Though we're also going to work with the [Nimble](https://wakoma.co/nimble/) project which lets communities run services locally on their mesh node which is similar to the Atek concept (but more manual at the moment).

Originally, I ran into Beaker Browser because I need to find a solution for my case. I have a remote farmland where the only possible communication is 4G network. I need to it send surveillance camera videos including other irrigation controls to my home (which is also run on 4G router). The idea of p2p inspired me and I have explored BB (without success) and many other stuffs. Recently I found Syncthing and it works in the sense that I can update files. I also explored IPFS and FreeNet but they seem to be too big for me. After seeing your reply and Agregore Browser I am learning more p2p from your blog.

**Comment by astrolox** [comment]
@pfrazee Hi, I've just stumbled across Beaker Browser. Based on the website the project looks alive, If it's now dead, perhaps you could archive the github repo and redirect/close the website? (to help avoid confusion)

**Reply by pfrazee** [reply]
Yeah, I'll get that done pretty soon actually
